contrib/plugins/helloworld.py

```diff
@@ -45,12 +45,27 @@ def json_getmanifest(request):
     }
 
 
+def plugin_log(message, level="info"):
+    payload = {
+        "jsonrpc": "2.0",
+        "method": "log",
+        "params": {
+            "level": level,
+            "message": message,
+        }
+    }
+    json.dump(payload, fp=sys.stdout)
+    sys.stdout.write('\n\n')
+    sys.stdout.flush()
+
+
 def json_init(request, options, configuration):
     """The main daemon is telling us the relevant cli options
     """
     global greeting
-
     greeting = request['params']['options']['greeting']
+    plugin_log("Plugin helloworld.py initialized with greeting \"{}\"".format(greeting), "debug")
```

Comment:
This allows the developer to depict a message via calling b plugin_log. This is very helpful and nice! For exceptions I will be able to catch them and push the error string to this method. 

However if my third party libs use std.out I will still run into problems.

Reply:
well that might not be to bad. for exaple in flask I could do app.run in this case. 
also I could regularly do plugin_log(f.getvalue()) to send the message to c-lightning) but yes I guess this is really just temporary.

Reply:
Working on a plugin class that monkey-patches stdout and stderr so we can start using these streams like we normally do, but `print`s result in log notifications instead.

Reply:
Note that stderr now goes to parent stderr: this is nice for simple debugging.  Like when my plugins backtrace :)

Reply:
My small python Plugin library grabs `stderr` as well and turns things into log notifications. This has the advantage that it provides more context than just spurious writes to `stderr`. See the `plugin-6` PR that I'll open in a few minutes.